### PR TITLE
Watson conversation

### PR DESCRIPTION
**Pull request by jkingoliver**
fyi https://github.com/IBM-Swift/CloudConfiguration/pull/25

### CODE REVIEW

**Comment by tunniclm** [comment]
Changed to be targeted against `develop`.

File: refresh/templates/services/watsonconversation/initializeBluemixService.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            
+                conversation = Conversation(username: username, password: password, version: version)
+        
+                conversation!.message(withWorkspace: workspaceID, failure: failure) { response in
+                    print(response.output.text)
+                    context = response.context
+            } 
+        }
+    }
+
```
Comment (tunniclm) [inline refresh/templates/services/watsonconversation/initializeBluemixService.swift]: Can this code be compressed? In other services we have used an extension of the client object to handle creation directly from the service returned by the ConfigurationManager.

File: refresh/templates/services/watsonconversation/importDependency.swift [hunk 2]
```diff
@@ -0,0 +1 @@
+        .Package(url: "https://github.com/watson-developer-cloud/swift-sdk",    majorVersion: 0),
```
Comment (tunniclm) [inline refresh/templates/services/watsonconversation/importDependency.swift:1]: For dependencies with major version at `0`, I like to specify the minor as well. Whilst there are no rules for `0` versions (the are counted as "in development"), often projects will update the minor on breaking changes.

There is also the question of whether anything should really be a `0` release if other projects are depending on it.

**Review by tunniclm — CHANGES_REQUESTED** [review]
This good work, just a couple of things I'd like to see either discussed / changed. If you disagree with any of the comments, I'd love to hear the reasons and discuss. Thanks :)

**Review by tunniclm — COMMENTED** [review]
Look like you will need to rebase this branch before we can merge.

File: refresh/templates/services/watsonconversation/importDependency.swift (hunk 2 above)
Comment (jkingoliver) [reply refresh/templates/services/watsonconversation/importDependency.swift:1]: As we discussed in scrum, we will address questions with how to represent dependency versions at our next sprint planning meeting

File: refresh/templates/services/watsonconversation/initializeBluemixService.swift (hunk 1 above)
Comment (jkingoliver) [reply refresh/templates/services/watsonconversation/initializeBluemixService.swift]: I extracted the initialization code to the Extension file and removed the sample code.

**Review by tunniclm — APPROVED** [review]

**Pull request merged**